zigup.zig

```diff
@@ -500,13 +512,26 @@ fn fetchDownloadIndex(allocator: Allocator, index_url: []const u8) !DownloadInde
     return DownloadIndex{ .text = text, .json = json };
 }
 
-fn loggyMakeDirAbsolute(dir_absolute: []const u8) !void {
-    if (builtin.os.tag == .windows) {
-        loginfo("mkdir \"{s}\"", .{dir_absolute});
-    } else {
-        loginfo("mkdir '{s}'", .{dir_absolute});
+fn loggyMakeDirAbsoluteParents(dir_absolute: []const u8) !void {
+    var itr = try std.fs.path.componentIterator(dir_absolute);
```

Comment:
I think we can just call makePathAbsolute here instead?

Reply:
Do you mean `std.fs.makeDirAbsolute()`?

That's what was called before, but because of the other changes, `makeDirAbsolute` will fail if the parent directory does not exist.  (A case that was basically impossible to hit with `~/zig`.)

Reply:
Can you call `makePath` first? https://github.com/ziglang/zig/blob/master/lib/std/fs/Dir.zig#L1140

Reply:
Ahh, I completely missed that API.  Thanks!

Sure, that's better than local interpretation.  I'm on it :)